ssl/ssl_asn1.c

```diff
@@ -316,6 +330,18 @@ SSL_SESSION *d2i_SSL_SESSION(SSL_SESSION **a, const unsigned char **pp,
     ret->peer = as->peer;
     as->peer = NULL;
 
+    EVP_PKEY_free(ret->peer_rpk);
+    ret->peer_rpk = NULL;
+    if (as->peer_rpk != NULL) {
+        const unsigned char *data = as->peer_rpk->data;
+
+        /*
+         * |data| is incremented; we don't want to lose original ptr
+         * No libctx/propq available, so use non-ex version
```

Comment:
Errr....no! We can't do that. We're going to need another solution. Probably we need a `d2i_SSL_SESSION_ex`. All internal callers of `d2i_SSL_SESSION` should be updated to use the `_ex` variant. We should test RPK works with a non-default libctx

Reply:
So, is that within scope of this PR, or is that another task?

Reply:
The change isn't very big, and I can do it as part of this PR (or at least as a separate commit).

Reply:
It's part of the fixup commits in this PR.